bazel/python_repositories.bzl

```diff
@@ -0,0 +1,44 @@
+# Copyright 2021 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""All external repositories used for Python dependencies"""
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+load("@rules_python//python:pip.bzl", "pip_parse")
+load("//bazel:python_packages.bzl", "PYTHON_PACKAGE")
+
+def python_repositories(name):
+    http_archive(
+        name = PYTHON_PACKAGE.name,
+        build_file_content = PYTHON_PACKAGE.build_file_content,
+        patch_cmds = PYTHON_PACKAGE.patch_cmds,
+        sha256 = PYTHON_PACKAGE.sha256,
+        strip_prefix = PYTHON_PACKAGE.strip_prefix,
+        urls = PYTHON_PACKAGE.urls,
+    )
+
+    native.register_toolchains("//bazel:py_toolchain")
+
+    pip_parse(
+        name = "python_deps",
+        extra_pip_args = ["--require-hashes"],
+        python_interpreter_target = "@python_interpreter//:python_bin",
+        requirements_lock = "//lte/gateway/python:requirements.txt",
+        visibility = ["//visibility:public"],
+    )
+
+    git_repository(
```

Comment:
So from my reading, `subpar` is now deprecated and we can achieve python binary generation with the `py_binary` rule.  Am I wrong? Would be nice to drop subpar use if this is correct.  I base this off my investigation of the prototype PR, which led me to read the Github page for subpar (where I noticed no commits since 2019). See [docs](https://github.com/google/subpar).

Reply:
Overall, there seems to be rather a lot of movement in the infrastructure of the project lately. And also seeing the latest efforts in dockerization (e.g. https://github.com/magma/magma/issues/10498), further persuing this path, we will get build artifacts in terms of a docker image, which are
* (mostly) self contained,
* well standardized,
* "multi-platform" of sorts, and
* of artifact type equal across all/most product components.

In prospect of this development we might want to openly (and orally?) discuss, if a hermetically bundled Python apart from docker images, is still reasonable effort or how alterative strategies for the build setup could look like.

More specifically to this PR, I think, the topic of hermetic builds would be out of scope here, and would be big enough to justify a separate step (and PR), in case of desireability. With this aspect, in mind, I think, we should actually remove the "par-stuff" completely from this PR, as it is not strictly required to achieve the original goal of it, i.e. "replace make with bazel". @ajahl

Reply:
@ajahl removed subpar. ✔️